src/System.Diagnostics.TextWriterTraceListener/tests/CommonUtilities.cs

```diff
@@ -1,88 +1,151 @@
-            {
-                bool found = false;
-                for (int j = 0; j < collection.Count; j++)
-                {
-                    if (collection[j].GetType().Equals(types[i]))
-                    {
-                        found = true;
-                        Assert.True(collection[j].Name == names[i]);
-                        if (filters[i] == null)
-                            Assert.True(collection[j].Filter == null);
-                        else
-                            Assert.True(collection[j].Filter.GetType().Equals(filters[i]));
-                    }
-                }
+            const string delimiter = ";";
+            var builder = new StringBuilder(EscapedString(source));
+            builder.Append(delimiter);
+            builder.Append(eventType.ToString());
+            builder.Append(delimiter);
+            builder.Append(id.ToString(CultureInfo.InvariantCulture));
+            builder.Append(delimiter);
+            builder.Append(EscapedString(args != null ? string.Format(format, args) : format));
+            builder.Append(delimiter);
+            builder.Append(delimiter);
 
-                Assert.True(found);
+            if (cache != null)
+            {
+                builder.Append(cache.ProcessId);
+                builder.Append(delimiter);
+                builder.Append(EscapedString(cache.ThreadId));
+                builder.Append(delimiter);
+                builder.Append(EscapedString(cache.DateTime.ToString("o", CultureInfo.InvariantCulture)));
+                builder.Append(delimiter);
+                builder.Append(cache.Timestamp.ToString(CultureInfo.InvariantCulture));
+                builder.Append(delimiter);
+            }
+            else
+            {
+                for (int i = 0; i < 5; ++i)
+                    builder.Append(delimiter);
             }
 
-            return true;
+            return builder.AppendLine().ToString();
         }
 
-        public static bool TestListenerContent(Type[] types, string[] names, string presentMsg)
+        internal static string ExpectedTraceDataOutput(TraceFilter filter, TraceEventCache cache, string source, TraceEventType eventType, int id, object data)
         {
-            return TestListenerContent(types, names, presentMsg, null);
+            if (filter != null && !filter.ShouldTrace(cache, source, eventType, id, null, null, data, null))
+                return string.Empty;
+
+            const string delimiter = ";";
+            var builder = new StringBuilder(EscapedString(source));
+            builder.Append(delimiter);
+            builder.Append(eventType.ToString());
+            builder.Append(delimiter);
+            builder.Append(id.ToString(CultureInfo.InvariantCulture));
+            builder.Append(delimiter);
+            builder.Append(delimiter);
+            builder.Append(EscapedString(data.ToString()));
+            builder.Append(delimiter);
+
+            if (cache != null)
```

Comment:
There's a lot of duplication between these two ExpectedTrace*Output functions.  Is it not possible to share most of the logic?  For example, the output from the cache section looks identical.